lib/router.js

```diff
@@ -46,10 +46,26 @@ export default class Router {
 
     this.historyMethod = historyMethod;
     this.origChangeStateMethod = this.window.history[this.historyMethod];
-    this.window.history[this.historyMethod] = (...args) => {
-      this.origChangeStateMethod.apply(this.window.history, args);
-      this.navigateToHash();
-    };
+
+    // If this is the true *original* native method, fire pushstate/replacestate events so
+    // that other components can be notified of history changes. Subsequent panel routers
+    // will re-wrap the method but not fire duplicate pushstate/replacestate events.
+    // Note: this.origChangeStateMethod.name === this.historyMethod is an effective means
+    // of verifying whether this.window.history[this.historyMethod] has been wrapped yet or
+    // not because the original native function will have name === "pushState" / "replaceState",
+    // but the wrapped version will have name === undefined since it is anonymous.
+    if (this.origChangeStateMethod.name === this.historyMethod) {
+      this.window.history[this.historyMethod] = (...args) => {
+        this.origChangeStateMethod.apply(this.window.history, args);
+        this.navigateToHash();
+        window.dispatchEvent(new CustomEvent(this.historyMethod.toLowerCase()));
```

Comment:
smol nit. I'd prefer we not lowercase the event name. pushState and replaceState feel more readable. We already do camelCase for custom events anyway.

Reply:
fair, the lowercase was for consistency with native ["popstate"](https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event), makes me a little sad that it's lowercase but wdyt?

Reply:
I guess that makes sense then. Let's leave as is. Perhaps a comment?